src/main/java/forum/entity/Thread.java

```diff
@@ -0,0 +1,84 @@
+package forum.entity;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+
+@Entity
+@Table(name="thread")
+public class Thread {
+
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	@Column(name = "id")
+	private int id;
+
+	@Column(name = "title")
+	private String title;
+	
+	@Column(name="content")
+	private String content;
+	
+	
+	@ManyToOne(cascade = { CascadeType.PERSIST, CascadeType.MERGE,
+							CascadeType.DETACH, CascadeType.REFRESH })
+	@JoinColumn(name = "category_id")
+	private Category category;
+	
+	public Thread() {
+		
+	}
+
+	public Thread(int id, String title) {
+		this.id = id;
+		this.title = title;
+	}
+
+	public int getId() {
+		return id;
+	}
+
+	public void setId(int id) {
+		this.id = id;
+	}
+
+	public String getTitle() {
+		return title;
+	}
+
+	public void setTitle(String title) {
+		this.title = title;
+	}
+	
+
+	public String getContent() {
+		return content;
+	}
+
+	public void setContent(String content) {
```

Comment:
Hmm do we need all these setters and getters?

Reply:
The setters for content and title don't seem to affect the json that the server return. So should I remove them then?